lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -719,36 +748,38 @@ finalizeModel(GeneratedCodeBasicInfo &GCBI,
           }
         }
 
-        if (FunctionEdgeType::isCall(Edge->Type)) {
-          auto *CE = llvm::cast<CallEdge>(Edge.get());
+        if (efa::FunctionEdgeType::isCall(Edge->Type)) {
+          auto *CE = llvm::cast<efa::CallEdge>(Edge.get());
           const auto IDF = Binary.ImportedDynamicFunctions;
           bool IsDynamicCall = (not SymbolName.empty()
                                 and IDF.count(SymbolName.str()) != 0);
 
           if (IsDynamicCall) {
             // It's a dynamic function call
-            revng_assert(CE->Type == model::FunctionEdgeType::FunctionCall);
+            revng_assert(CE->Type == efa::FunctionEdgeType::FunctionCall);
             CE->Destination = MetaAddress::invalid();
             CE->DynamicFunction = SymbolName.str();
 
             // The prototype is implicitly the one of the callee
-            revng_assert(not CE->Prototype.isValid());
+            revng_assert(not Function->CallSitePrototype.count(Block.Start));
           } else if (CE->Destination.isValid()) {
             // It's a simple direct function call
-            revng_assert(CE->Type == model::FunctionEdgeType::FunctionCall);
+            revng_assert(CE->Type == efa::FunctionEdgeType::FunctionCall);
 
             // The prototype is implicitly the one of the callee
-            revng_assert(not CE->Prototype.isValid());
+            revng_assert(not Function->CallSitePrototype.count(Block.Start));
           } else {
             // It's an indirect call: forge a new prototype
             auto Prototype = buildPrototype(GCBI, Binary, Summary, Block);
-            CE->Prototype = Binary.recordNewType(std::move(Prototype));
+            auto TypedPrototype = Binary.recordNewType(std::move(Prototype));
+            Function->CallSitePrototype.insert({ Block.Start, TypedPrototype });
           }
         }
       }
     }
 
-    Function->CFG = Summary.CFG;
+    efa::FunctionMetadata FM(Function->Entry, Summary.CFG);
+    FM.verify(Binary);
```

Comment:
You're ignoring the return value.
Usually we have an overload with a `bool` stating whether you should assert, if you use that this is fine.